app/src/main/java/com/firebase/uidemo/ChooserActivity.java

```diff
@@ -31,73 +34,99 @@
 
 import butterknife.BindView;
 import butterknife.ButterKnife;
-import butterknife.OnItemClick;
 
 public class ChooserActivity extends AppCompatActivity {
-
-    private static final Class[] CLASSES = new Class[]{
-            ChatActivity.class,
-            AuthUiActivity.class,
-            ImageActivity.class,
-    };
-
-    private static final int[] DESCRIPTION_NAMES = new int[] {
-            R.string.name_chat,
-            R.string.name_auth_ui,
-            R.string.name_image
-    };
-
-    private static final int[] DESCRIPTION_IDS = new int[] {
-            R.string.desc_chat,
-            R.string.desc_auth_ui,
-            R.string.desc_image
-    };
-
-    @BindView(R.id.list_view)
-    ListView mListView;
+    @BindView(R.id.activities)
+    RecyclerView mActivities;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_chooser);
         ButterKnife.bind(this);
 
-        mListView.setAdapter(new MyArrayAdapter(
-                this,
-                android.R.layout.simple_list_item_2,
-                CLASSES));
+        mActivities.setLayoutManager(new LinearLayoutManager(this));
+        mActivities.setAdapter(new ActivityChooserAdapter());
+        mActivities.setHasFixedSize(true);
     }
 
-    @OnItemClick(R.id.list_view)
-    public void onItemClick(int position) {
-        Class clicked = CLASSES[position];
-        startActivity(new Intent(this, clicked));
-    }
+    private static class ActivityChooserAdapter extends RecyclerView.Adapter<ActivityStarterHolder> {
+        private static final Class[] CLASSES = new Class[]{
+                ChatActivity.class,
+                AuthUiActivity.class,
+                ImageActivity.class,
+        };
+
+        private static final int[] DESCRIPTION_NAMES = new int[]{
+                R.string.name_chat,
+                R.string.name_auth_ui,
+                R.string.name_image
+        };
+
+        private static final int[] DESCRIPTION_IDS = new int[]{
+                R.string.desc_chat,
+                R.string.desc_auth_ui,
+                R.string.desc_image
+        };
 
-    public static class MyArrayAdapter extends ArrayAdapter<Class> {
-
-        private Context mContext;
+        @Override
+        public ActivityStarterHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View itemView = LayoutInflater.from(parent.getContext())
+                    .inflate(R.layout.activity_chooser_item, parent, false);
+
+            // Set custom margins
+            int marginsInPx = (int) TypedValue.applyDimension(
```

Comment:
I'm [including `android.R.layout.simple_list_item_2`](https://github.com/firebase/FirebaseUI-Android/pull/610/files#diff-a6396a78e0409ffd093003e115647a15R11) which doesn't let me set margins in XML. If you'd like, we can use our own custom layout instead of the native one. I just felt that it was easier to do this in code rather than copying the XML, but I'm fine with it either way.